### ISSUE TITLE
Using Android Adreno drivers with LibHybris and Wayland

### ISSUE THREAD

**Issue by depau** [crondog/arch-flo#8]
I just saw these:
- http://www.cnx-software.com/2013/04/21/how-to-use-libhybris-and-android-gpu-libraries-with-mer-linux-on-the-cubieboard/
- http://archlinuxarm.org/forum/viewtopic.php?f=33&t=5951

I think we can do it. I'm building the CM source right now. I'll tell you if it works.

**Comment by willcast** [comment]
Did it work?

**Comment by depau** [comment]
I'm trying right now. Last time I tried my Android source was stored in a 60 GB ext4 filesystem which wasn't enough. Now I converted it to btrfs, racked up around 90 more GBs from OSes I'm not using and added them to the FS. I also enabled compression, so I should be good for a while, even with ccache enabled.

I'm going to start the build now and see what happens. It's going to take a lot of time as I don't have a fast CPU, it doesn't even have HyperThreading...

**Comment by depau** [comment]
Luckily I haven't started... I just read this: http://linux-sunxi.org/CedarX/libve#Using_libhybris_wrapped_Android_libraries

According to that page, you just need to have _an_ Android /system, and we do, we're trying to dual-boot Android and Arch (or at least I am). I'm not sure whether the bionic patches are actually needed. I'll try without them, and if it doesn't work, I'll build patched CyanogenMod.
We only need the headers, and if my definition of headers isn't wrong, we have them in the source, which I already downloaded.

Moreover the patched hybris in that page has video decoding support, which is kind of sweet.

**Comment by bogsen** [comment]
fwiw, you probably don't want compression enabled if you're constrained by your CPU as it is

**Comment by depau** [comment]
You're right, it's a good idea. I have an Intel Pentium Dual-Core T4400 CPU, it's not that great...

**Comment by depau** [comment]
Found an updated guide with updated patches (the page I linked earlier had patches that don't work anymore with current bionic, as they ported some stuff from c to c++): http://linux-rockchip.info/mw/index.php?title=LibHybris

By the way, it turns out the bionic /system patch is needed. I tried to mount it and when trying to run a libhybris test it segfaults. Now I wonder if it's possible to run Android with that patch - I mean, if I patch bionic, build the rom and install it, will it boot normally?

**Comment by depau** [comment]
I forgot to update you about the status of my tests: I finally managed to build CyanogenMod with the bionic patches; I had to use Ubuntu, I would only get ld bus errors on Arch Linux (which is dead now, I can't boot it because of a systemd problem; I'll be moving back to Ubuntu at the next issue, in the case I get it working again).
I can't, however, run the tests: I get segfaults and similar errors for most of the tests, especially for the ones regarding GL. I [opened a new issue](https://github.com/libhybris/libhybris/issues/245) in LibHybris. I'll also see if Ubuntu uploaded some patches, they must have gotten it working if Ubuntu Touch works. I'll check when I have time.
